src/sparsematrix.jl

```diff
@@ -3908,27 +3908,24 @@ function vcat(X::AbstractSparseMatrixCSC...)
         ptr_res = colptr[c]
         for i = 1 : num
             colptrXi = getcolptr(X[i])
-            col_length = (colptrXi[c + 1] - 1) - colptrXi[c]
+            rowvalXi = rowvals(X[i])
```

Comment:
I think at some point we agreed not to create temporary variables for this use `rowvals(X[i])` directly it gets optimized away by the compiler

Reply:
Addressed by b961a7f. I kept the `getcolptr` stuff in `vcat` as it was before, to avoid touching lines unnecessarily. Let me know if you disagree and would like it changed by this PR.

Reply:
since it's jus style, let's keep the pull request focused on just `repeat`